x-pack/plugin/relevance-search/src/test/java/org/elasticsearch/xpack/relevancesearch/query/RelevanceMatchQueryRewriterTests.java

```diff
@@ -0,0 +1,243 @@
+            .filter(
+                new BoolQueryBuilder().mustNot(
+                    new BoolQueryBuilder().must(new TermsQueryBuilder("_index", HIDDEN_INDEX))
+                        .must(new TermsQueryBuilder("_id", HIDDEN_DOC_1_ID))
+                )
+                    .mustNot(
+                        new BoolQueryBuilder().must(new TermsQueryBuilder("_index", HIDDEN_INDEX))
+                            .must(new TermsQueryBuilder("_id", HIDDEN_DOC_2_ID))
+                    )
+            )
+            .toQuery(context);
+
+        assertEquals(expectedQuery, obtainedQuery);
+    }
+
+    public void testQueryWithCurationsNoHiddenDocs() throws IOException, CurationsService.CurationsSettingsNotFoundException {
+
+        final String queryText = "matching";
+        queryBuilder.setQuery(queryText);
+
+        setRelevanceSettings();
+        setCurationsSettings(true, false, queryText);
+
+        SearchExecutionContext context = createSearchExecutionContext();
+        Query obtainedQuery = queryRewriter.rewriteQuery(queryBuilder, context);
+
+        Query expectedQuery = new PinnedQueryBuilder(
+            new CombinedFieldsQueryBuilder(queryText, RELEVANCE_SETTINGS_FIELDS),
+            new PinnedQueryBuilder.Item(PINNED_DOC_1_ID, PINNED_INDEX),
+            new PinnedQueryBuilder.Item(PINNED_DOC_2_ID, PINNED_INDEX)
+        ).toQuery(context);
+
+        assertEquals(expectedQuery, obtainedQuery);
+    }
+
+    public void testQueryWithCurationsNoPinnedDocs() throws IOException, CurationsService.CurationsSettingsNotFoundException {
+
+        final String queryText = "matching";
+        queryBuilder.setQuery(queryText);
+
+        setRelevanceSettings();
+        setCurationsSettings(false, true, queryText);
+
+        SearchExecutionContext context = createSearchExecutionContext();
+        Query obtainedQuery = queryRewriter.rewriteQuery(queryBuilder, context);
+
+        Query expectedQuery = new BoolQueryBuilder().should(new CombinedFieldsQueryBuilder(queryText, RELEVANCE_SETTINGS_FIELDS))
+            .filter(
+                new BoolQueryBuilder().mustNot(
+                    new BoolQueryBuilder().must(new TermsQueryBuilder("_index", HIDDEN_INDEX))
+                        .must(new TermsQueryBuilder("_id", HIDDEN_DOC_1_ID))
+                )
+                    .mustNot(
+                        new BoolQueryBuilder().must(new TermsQueryBuilder("_index", HIDDEN_INDEX))
+                            .must(new TermsQueryBuilder("_id", HIDDEN_DOC_2_ID))
+                    )
+            )
+            .toQuery(context);
+
+        assertEquals(expectedQuery, obtainedQuery);
+    }
+
+    public void testQueryWithCurationDoesNotMatchQuery() throws IOException, CurationsService.CurationsSettingsNotFoundException {
+
+        final String queryText = "matching";
+        queryBuilder.setQuery(queryText);
+
+        setRelevanceSettings();
+        setCurationsSettings(true, true, "query that will not match");
```

Comment:
Oh, interesting! So for curations, do we never perform analysis (things like stemming) on curated query matches? (Don't mind me, this is just for my information, not a review nit).

Reply:
Right now we don't - although the previous comments on normalizing values might have to do with it. We probably want to use the analyzers here, but I'm uncertain of the consequences of that.